### PR TITLE
getBoundingClient should return a DOMRect not a ClientRect

### PR DESCRIPTION
**Pull request by ryota-murakami**
## Introduction
@gregtatum is already noticed this issue at #5475, and his explain is still correct today.

but above issue was closed without fix PR.
So I created PR along to #5475 reports.

Thanks any review😀

### CODE REVIEW

File: lib/dom.js [hunk 1]
```diff
@@ -3568,7 +3568,35 @@ declare class ClientRectList { // extension
   [index: number]: ClientRect;
 }
 
-// TODO: HTML*Element
```
Comment (motiz88) [inline lib/dom.js]: I'm not a maintainer here, but I think this `// TODO` comment should probably be kept - since this PR doesn't actually address it.

Comment (ryota-murakami) [reply lib/dom.js]: @motiz88 Thanks your review.
Yeah, I agree with you and I've  pushed revert commit.

File: lib/dom.js [hunk 2]
```diff
@@ -3568,6 +3568,36 @@ declare class ClientRectList { // extension
...
+declare class DOMRectReadOnly {
+  +bottom: number;
+  +height: number;
+  +left: number;
+  +right: number;
+  +top: number;
+  +width: number;
+  +x: number;
+  +y: number;
+  fromRect(rectangle?: {x: number, y: number, width: number, height: number}): DOMRect;
```
Comment (mroch) [inline lib/dom.js]: couple of things:

1) should be `static`
2) if i'm reading [the draft](https://drafts.fxtf.org/geometry/#domrectreadonly) correctly, `DOMRectReadOnly.fromRect()` should return a `DOMRectReadOnly`, and `DOMRect.fromRect()` should return a `DOMRect`. so add one to DOMRect as well

File: lib/dom.js [hunk 3]
```diff
@@ -3568,6 +3568,36 @@ declare class ClientRectList { // extension
   [index: number]: ClientRect;
 }
 
+declare class DOMRectReadOnly {
```
Comment (mroch) [inline lib/dom.js]: missing the constructor: https://drafts.fxtf.org/geometry/#dom-domrectreadonly-domrectreadonly

File: lib/dom.js [hunk 4]
```diff
@@ -3568,6 +3568,38 @@ declare class ClientRectList { // extension
...
```
Comment (fishythefish) [inline lib/dom.js]: Seems like these classes are already declared in [`bom.js`](https://github.com/facebook/flow/blob/master/lib/bom.js#L395)?

Comment (ryota-murakami) [reply lib/dom.js]: Oh I didn't realize it.
I see, surely that's not DOM things BOM is property place.

next my plan is 
1. fix Exists DOMRect definition based on @mroch review.
2. add DOMRect(already declared in bom.js) to getBoundingClient() return type.

Comment (ryota-murakami) [reply lib/dom.js]: @fishythefish I moved these classes to bom.js!

**Pull request closed without merging**